### PR TITLE
No manual lib file deps globs

### PR DESCRIPTION
**Pull request by rgrinberg**
* Fixes a bug when installing private cmi's
* Uses automatic globs for setting up lib artfiact deps. As a consequence, this should make the virtual lib copying rules correct in some instances where the virtual lib is reinstalled and the implementation is local. Although it is too hard to create a test that showcases this.

### CODE REVIEW

File: test/blackbox-tests/test-cases/private-modules/run.t [hunk 1]
```diff
@@ -9,10 +9,11 @@
   File "foo.ml", line 1, characters 0-5:
   Error: Unbound module X
 
-  $ dune build --root excluded-from-install-file | grep -i priv
+Private modules are not excluded from the install file, but installed in the .private subdir
+  $ dune build --root private-subdir | grep -i priv
   Entering directory 'excluded-from-install-file'
```
Comment (aalekseyev) [inline test/blackbox-tests/test-cases/private-modules/run.t]: "Entering directory 'excluded-from-install-file'" message seems out of date?

**Comment by aalekseyev** [comment]
I'm not going to pretend that I understand the old code in detail, but the new code is at least simple enough.

My concern is: is it possible for multiple libraries to share the same `obj_lib`? If so, then the glob will overapproximate dependencies, and then potentially you have dependency cycles or building too much.

**Comment by aalekseyev** [comment]
After digging through what feels like a few dozen of indirections I got to this definition in `src/utils.ml`:

```
let library_object_directory ~dir name =
  Path.relative dir ("." ^ Lib_name.Local.to_string name ^ ".objs")
```

It looks like no, multiple libraries can't share the same obj_dir.

**Comment by rgrinberg** [comment]
Yes it is impossible. Moreover, executables get their own obi dir as well

**Comment by diml** [comment]
It's nice that we simplified this code! The old code required to carefully maintain these aliases and was a bit of a pain

**Pull request merged**